lib/p2p/Pool.ts

```diff
@@ -96,26 +107,11 @@ class Pool extends EventEmitter {
     }
 
     if (this.server) {
-      let externalIp: string | undefined;
-      // Fetch the external IP if no address was specified by the user
-      if (this.addresses.length === 0) {
-        try {
-          externalIp = await getExternalIp();
-
-          this.logger.info(`retrieved external IP: ${externalIp}`);
-        } catch (error) {
-          this.logger.error(error.message);
-        }
-      }
-
       await this.listen();
       this.bindServer();
 
-      if (externalIp) {
-        this.addresses.push({
-          host: externalIp,
-          port: this.listenPort!,
-        });
+      if (this.config.detectexternalip && this.addresses.length === 0) {
```

Comment:
Let's drop the `length === 0` check here.

Reply:
sure thing